astropy/cosmology/funcs/optimize.py

```diff
@@ -34,16 +34,16 @@ def _z_at_scalar_value(
     """
     from scipy.optimize import minimize_scalar
 
-    opt = {"maxiter": maxfun}
+    opt = {"maxiter": maxfun, "xtol": ztol}
     # Assume custom methods support the same options as default; otherwise user
     # will see warnings.
-    if str(method).lower() == "bounded":
-        opt["xatol"] = ztol
+    if callable(method):  # can skip callables
+        pass
+    elif str(method).lower() == "bounded":
```

Comment:
Why make it case-insensitive here?

Reply:
It was already case insensitive.

```
if str(method).lower() == "bounded":
        opt["xatol"] = ztol
```

are the lines this is replacing.

Reply:
Just curious, what was the original motivation to make it case insensitive in the first place?

Reply:
Though it isn't documented, the scipy implementation is case insensitive!

https://github.com/scipy/scipy/blob/de80faf9d3480b9dbb9b888568b64499e0e70c19/scipy/optimize/_minimize.py#L863